### PR TITLE
[SDL-0266] Feature/New vehicle data `GearStatus`

### PR DESCRIPTION
**Pull request by Mariia-Novosiadla**
Guidelines updates according to issue [#3216](https://github.com/smartdevicelink/sdl_core/issues/3216).

Proposal [SDL-0266](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0266-New-vehicle-data-GearStatus.md)

This PR is **ready** for review.

### Summary

1.  Updates for **Common**
* extending `VehicleDataType` with new element `VEHICLEDATA_GEARSTATUS`
* adding new elements to `PRNDL` enum
* adding new enum `TransmissionType`
* adding new struct `GearStatus`

2. Updates in **VehicleInfo**
* adding `gearStatus` to request/response params list of the following RPCs:
SubscribeVehicleData
UnsubscribeVehicleData
GetVehicleData
OnVehicleData

### CLA
* [x]  I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)s

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
PR Approved.

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

File: docs/Common/Enums/index.md [hunk 1]
```diff
@@ -925,7 +926,7 @@
 |PARK|0|Parking|
 |REVERSE|1|Reverse gear|
 |NEUTRAL|2|No gear|
-|DRIVE|3||
+|DRIVE|3|Regula Drive mode|
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:929]:
```suggestion
|DRIVE|3|Regular Drive mode|
```

File: docs/Common/Enums/index.md [hunk 2]
```diff
@@ -889,6 +889,7 @@
 |VEHICLEDATA_ELECTRONICPARKBRAKESTATUS|29||
 |VEHICLEDATA_CLOUDAPPVEHICLEID|30| Parameter used by cloud apps or the policy server to identify a head unit|
 |VEHICLEDATA_OEM_CUSTOM_DATA|31||
+|VEHICLEDATA_GEARSTATUS|32||
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:892]: The enum value should match the position in the HMI_API spec https://github.com/LuxoftSDL/sdl_core/blob/feature/0266_new_vehicle_data_gear_status/src/components/interfaces/HMI_API.xml#L1174

Gear Status should be placed before PRNDL
```md
|VEHICLEDATA_GEARSTATUS|8||
```
and the subsequent enum values should be incremented.
```md
|VEHICLEDATA_PRNDL|9||
|VEHICLEDATA_TIREPRESSURE|10||
|VEHICLEDATA_ODOMETER|11||
|VEHICLEDATA_BELTSTATUS|12||
...
```

File: docs/Common/Enums/index.md [hunk 3]
```diff
@@ -1303,3 +1306,16 @@
...
+|Name|Value|Description|
+|:---|:----|:----------|
+|MANUAL|0|Manual transmission.|
+|AUTOMATIC|1|Automatic transmission.|
+|SEMI_AUTOMATIC|2|Semi automatic transmission.|
+|DUAL_CLUTCH|3|Dual clutch transmission.|
+|CONTINUOUSLY_VARIABLE|4|Continuously variable transmission(CVT).|
+|INFINITELY_VARIABLE|5|Infinitely variable transmission.|
+|ELECTRIC_VARIABLE|6|Electric variable transmission.|
+|DIRECT_DRIVE|7|Direct drive between engine and wheels.|
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:1321]:
```suggestion
|DIRECT_DRIVE|7|Direct drive between engine and wheels.|

```

File: docs/Common/Structs/index.md [hunk 4]
```diff
@@ -1130,3 +1130,9 @@ There are no defined parameters for this struct
 |hybridAppPreference|[Common.HybridAppPreference](../enums/#hybridapppreference)|false||Specifies the user preference to use one specific app type or all available types|
 |endpoint|String|false|maxlength: 65535|If specified, which Core uses a client implementation of the connection type and attempts to connect to the endpoint when this app is selected (activated).<br>If omitted, Core won't attempt to connect as the app selection (activation) is managed outside of Core. Instead it uses a server implementation of the connection type and expects the app to connect|
 
+### GearStatus
+|Name|Type|Mandatory|Additional|Description|
+|:---|:---|:--------|:---------|:----------|
+|userSelectedGear|[Common.PRNDL](../enums/#prndl)|false||Gear position selected by the user<br>i.e. Park, Drive, Reverse|
+|actualGear|[Common.PRNDL](../enums/#prndl)|false||Actual Gear in use by the transmission|
+|transmissionType|[Common.TransmissionType](../enums/#transmissiontype)|false||Tells the transmission type|
```
Comment (ShobhitAd) [inline docs/Common/Structs/index.md:1138]:
```suggestion
|transmissionType|[Common.TransmissionType](../enums/#transmissiontype)|false||Tells the transmission type|

```

File: docs/Common/Enums/index.md (hunk 2 above)
Comment (Mariia-Novosiadla) [reply docs/Common/Enums/index.md]: fixed in f87bcddf47dfcb1763618e5d8fdadfd8a17966b5

File: docs/Common/Enums/index.md (hunk 3 above)
Comment (Mariia-Novosiadla) [reply docs/Common/Enums/index.md]: fixed in f87bcddf47dfcb1763618e5d8fdadfd8a17966b5

File: docs/Common/Enums/index.md [hunk 5]
```diff
@@ -865,30 +865,32 @@
...
+|VEHICLEDATA_ENGINETORQUE|24||
+|VEHICLEDATA_ACCPEDAL|25||
+|VEHICLEDATA_STEERINGWHEEL|26||
+|VEHICLEDATA_TURNSIGNAL|27||
+|VEHICLEDATA_FUELRANGE|28||
+|VEHICLEDATA_ENGINEOILLIFE|29||
+|VEHICLEDATA_ELECTRONICPARKBRAKESTATUS|30||
+|VEHICLEDATA_CLOUDAPPVEHICLEID|31| Parameter used by cloud apps or the policy server to identify a head unit|
+|VEHICLEDATA_OEM_CUSTOM_DATA|32||
+
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:893]:
```suggestion
|VEHICLEDATA_OEM_CUSTOM_DATA|32||
```
Just to follow the convention for other sections

File: docs/Common/Structs/index.md [hunk 6]
```diff
@@ -1130,3 +1130,9 @@ There are no defined parameters for this struct
 |hybridAppPreference|[Common.HybridAppPreference](../enums/#hybridapppreference)|false||Specifies the user preference to use one specific app type or all available types|
 |endpoint|String|false|maxlength: 65535|If specified, which Core uses a client implementation of the connection type and attempts to connect to the endpoint when this app is selected (activated).<br>If omitted, Core won't attempt to connect as the app selection (activation) is managed outside of Core. Instead it uses a server implementation of the connection type and expects the app to connect|
 
+### GearStatus
```
Comment (ShobhitAd) [inline docs/Common/Structs/index.md:1133]:
```suggestion
### GearStatus

```

File: docs/Common/Enums/index.md (hunk 5 above)
Comment (Mariia-Novosiadla) [reply docs/Common/Enums/index.md]: fixed in 78995a7a3f364d0a93ec57cb1b5b59c0099dffb6

File: docs/Common/Structs/index.md (hunk 6 above)
Comment (Mariia-Novosiadla) [reply docs/Common/Structs/index.md:1133]: fixed in 78995a7a3f364d0a93ec57cb1b5b59c0099dffb6

File: docs/Common/Enums/index.md [hunk 7]
```diff
@@ -1305,10 +1308,25 @@
 |CLOUD|1||
 |BOTH|2||
 
+<<<<<<< HEAD
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:1311]:
```suggestion

```

File: docs/Common/Enums/index.md [hunk 8]
```diff
@@ -1305,10 +1308,25 @@
...
+|MANUAL|0|Manual transmission.|
+|AUTOMATIC|1|Automatic transmission.|
+|SEMI_AUTOMATIC|2|Semi automatic transmission.|
+|DUAL_CLUTCH|3|Dual clutch transmission.|
+|CONTINUOUSLY_VARIABLE|4|Continuously variable transmission(CVT).|
+|INFINITELY_VARIABLE|5|Infinitely variable transmission.|
+|ELECTRIC_VARIABLE|6|Electric variable transmission.|  
+|DIRECT_DRIVE|7|Direct drive between engine and wheels.|
+=======
 ### CapacityUnit
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:1325]:
```suggestion

### CapacityUnit
```

File: docs/Common/Enums/index.md [hunk 9]
```diff
@@ -1305,10 +1308,25 @@
...
 
 |Name|Value|Description|
 |:---|:----|:----------|
 |LITERS|0||
 |KILOWATTHOURS|1||
 |KILOGRAMS|2||
+>>>>>>> origin/develop
```
Comment (ShobhitAd) [inline docs/Common/Enums/index.md:1332]:
```suggestion
|KILOGRAMS|2||

```

File: docs/Common/Enums/index.md (hunk 7 above)
Comment (AKalinich-Luxoft) [reply docs/Common/Enums/index.md]: @ShobhitAd fixed in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/214/commits/d20c6eb18a1a17d0b0da563deb346b018ff6fdd5

File: docs/Common/Enums/index.md (hunk 8 above)
Comment (AKalinich-Luxoft) [reply docs/Common/Enums/index.md]: @ShobhitAd fixed in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/214/commits/d20c6eb18a1a17d0b0da563deb346b018ff6fdd5

File: docs/Common/Enums/index.md (hunk 9 above)
Comment (AKalinich-Luxoft) [reply docs/Common/Enums/index.md]: @ShobhitAd fixed in https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/214/commits/d20c6eb18a1a17d0b0da563deb346b018ff6fdd5

**Pull request merged**